PyInstaller/utils/hooks/__init__.py

```diff
@@ -496,12 +490,26 @@ def django_dottedstring_imports(django_root_dir):
     # Many times Django users do not specify absolute imports in the settings module.
     pths.append(django_root_dir)
 
-    package_name = os.path.basename(django_root_dir) + '.settings'
-    env = {'DJANGO_SETTINGS_MODULE': package_name,
-           'PYTHONPATH': os.pathsep.join(pths)}
-    ret = eval_script('django_import_finder.py', env=env)
+    python_path = os.getenv('PYTHONPATH', '').split(os.pathsep)
+    python_path.extend(pths)
+    python_path = os.pathsep.join(python_path)
+
+    django_settings_module = os.getenv('DJANGO_SETTINGS_MODULE', os.path.basename(django_root_dir) + '.settings')
+    if env is None:
+        env = {}
+    env.setdefault('DJANGO_SETTINGS_MODULE', django_settings_module)
+    env.setdefault('PYTHONPATH', python_path)
```

Comment:
A normal assignment, would overwrite my original PYTHONPATH and I wanted to prevent that. My goal was to have better control of PYTHONPATH and DJANGO_SETTINGS_MODULE and let the automatic discovery do its thing only in cases when those env vars are undefined.